tools/clang/lib/Sema/SemaDecl.cpp

```diff
@@ -9818,7 +9818,11 @@ void Sema::CheckCompleteVariableDeclaration(VarDecl *var) {
       ActiveTemplateInstantiations.empty()) {
     PragmaStack<StringLiteral *> *Stack = nullptr;
     int SectionFlags = ASTContext::PSF_Implicit | ASTContext::PSF_Read;
-    if (var->getType().isConstQualified())
+    if (var->getType().isConstQualified()
+      // HLSL Change: for HLSL, const initialized isn't the same, unless static
+        && (!getLangOpts().HLSL ||
+            var->getTLSKind() == VarDecl::TLS_Static)
```

Comment:
We don't want isConstQualified() to return false, since we want other characteristics of const to still be applied elsewhere.  This logic also only applies to global constants (constant buffer values).  So is not generally applicable to the type.

Reply:
Ah, ok; then it makes sense.